ng2-components/ng2-activiti-tasklist/src/components/activiti-tasklist.component.ts

```diff
@@ -83,9 +83,21 @@ export class ActivitiTaskList implements OnInit, OnChanges {
         }
     }
 
-    ngOnInit() {
+    ngAfterContentInit() {
+        let schema: DataColumn[] = [];
+
+        if (this.columnList && this.columnList.columns && this.columnList.columns.length > 0) {
+            schema = this.columnList.columns.map(c => <DataColumn> c);
+        }
+
         if (!this.data) {
-            this.data = this.initDefaultSchemaColumns();
+            this.data = new ObjectDataTableAdapter([], schema.length > 0 ? schema : this.defaultSchemaColumn);
+        } else {
+            if (schema && schema.length > 0) {
```

Comment:
Could be better wrap this code in a method that explains what it does.
In addition, the method initDefaultSchemaColumns isn't used anymore, can we remove it?